SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -79,22 +247,65 @@ void ASpatialDebugger::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& Out
 	DOREPLIFETIME_CONDITION(ASpatialDebugger, WorkerRegions, COND_SimulatedOnly);
 }
 
-void ASpatialDebugger::Tick(float DeltaSeconds)
+void ASpatialDebugger::BeginPlay()
 {
-	Super::Tick(DeltaSeconds);
+	Super::BeginPlay();
 
 	check(NetDriver != nullptr);
 
+	if (ensureMsgf(!NetDriver->SpatialDebuggerSystem.IsValid(), TEXT("SpatialDebugger system is already created at BeginPlay")))
```

Comment:
I think this logic should all be on the netdriver, but trigger from the spatial debugger. You could also wrap the custom logic that's done in the constructor for clients in that functionality as well. And this call should be idempotent, because although we don't want to have more than a single spatial debugger in a world, there's current bugs where that can happen.